staging/src/k8s.io/client-go/tools/clientcmd/config.go

```diff
@@ -160,9 +160,12 @@ func ModifyConfig(configAccess ConfigAccess, newConfig clientcmdapi.Config, rela
 	// sort the possible kubeconfig files so we always "lock" in the same order
 	// to avoid deadlock (note: this can fail w/ symlinks, but... come on).
 	sort.Strings(possibleSources)
+
+	var lockErrors map[string]error = make(map[string]error)
+
 	for _, filename := range possibleSources {
 		if err := lockFile(filename); err != nil {
-			return err
+			lockErrors[filename] = err
 		}
 		defer unlockFile(filename)
```

Comment:
put this in an else block, we don't want to unlock files we did not actually lock

Reply:
Fixed in ecf3b83e88ef4c4791889f922514cf459d3b3058